src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx

```diff
@@ -0,0 +1,337 @@
+  getConnectorRef,
+  formatOnSubmitData
+} from './TerraformArtifactoryFormHelper'
+import type { PathInterface } from '../TerraformInterfaces'
+import stepCss from '@pipeline/components/PipelineSteps/Steps/Steps.module.scss'
+
+import css from './TerraformVarfile.module.scss'
+
+interface TFRemoteProps {
+  onSubmitCallBack: (data: any, prevStepData?: any) => void
+  isConfig: boolean
+  isTerraformPlan: boolean
+}
+
+export const TFArtifactoryForm: React.FC<StepProps<any> & TFRemoteProps> = ({
+  previousStep,
+  prevStepData,
+  onSubmitCallBack,
+  isConfig,
+  isTerraformPlan
+}) => {
+  const [selectedRepo, setSelectedRepo] = useState('')
+  const [connectorRepos, setConnectorRepos] = useState<SelectOption[]>()
+  const [filePathIndex, setFilePathIndex] = useState<{ index: number; path: string }>()
+  const [artifacts, setArtifacts] = useState<any>()
+  const { accountId, projectIdentifier, orgIdentifier } = useParams<{
+    projectIdentifier: string
+    orgIdentifier: string
+    accountId: string
+  }>()
+  const { getString } = useStrings()
+  const { showError } = useToaster()
+  const initialValues = formatInitialValues(isConfig, prevStepData, isTerraformPlan)
+  const connectorRef = getConnectorRef(isConfig, isTerraformPlan, prevStepData)
+
+  const {
+    data: ArtifactRepoData,
+    loading: ArtifactRepoLoading,
+    refetch: getArtifactRepos,
+    error: ArtifactRepoError
+  } = useGetRepositoriesDetailsForArtifactory({
+    queryParams: {
+      connectorRef: connectorRef,
+      accountIdentifier: accountId,
+      orgIdentifier,
+      projectIdentifier
+    },
+    lazy: true
+  })
+  const {
+    data: Artifacts,
+    loading: ArtifactsLoading,
+    refetch: GetArtifacts,
+    error: ArtifactsError
+  } = useGetArtifactsBuildsDetailsForArtifactory({
+    queryParams: {
+      connectorRef: connectorRef,
+      accountIdentifier: accountId,
+      orgIdentifier,
+      projectIdentifier,
+      repositoryName: selectedRepo,
+      maxVersions: 50,
+      filePath: filePathIndex?.path
+    },
+    debounce: 500,
+    lazy: true
+  })
+
+  useEffect(() => {
+    const repoName =
+      prevStepData?.varFile?.spec?.store?.spec?.repositoryName ||
+      prevStepData.formValues?.spec?.configuration?.configFiles?.store?.spec?.repositoryName ||
+      prevStepData.formValues?.spec?.configuration?.spec?.configFiles?.store?.spec?.repositoryName
+    if (repoName) {
+      setSelectedRepo(repoName)
+    }
+    const prevConfig = prevStepData.formValues?.spec?.configuration
+    const prevArtifacts = isTerraformPlan
+      ? prevConfig?.configFiles?.store?.spec?.artifacts
+      : prevConfig?.spec?.configFiles?.store?.spec?.artifacts
+    if (isConfig && prevArtifacts) {
+      setFilePathIndex({ index: 0, path: prevArtifacts[0].artifactFile.artifactPathExpression })
+    }
+  }, [prevStepData])
+
+  useEffect(() => {
+    if (ArtifactRepoError) {
+      showError(ArtifactRepoError.message)
+    }
+  }, [ArtifactRepoError])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'showError'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232248)
```suggestion
  }, [ArtifactRepoError, showError])
```